src/main/java/org/sharedid/endpoint/config/RouteConfiguration.java

```diff
@@ -0,0 +1,123 @@
+
+import java.util.List;
+
+@Configuration
+public class RouteConfiguration {
+    private static final Logger logger = LoggerFactory.getLogger(RouteConfiguration.class);
+
+    @Value("${templates.opt-out}")
+    private String optOutTemplate;
+
+    @Value("${templates.opt-in}")
+    private String optInTemplate;
+
+    @Bean
+    public HandlerRegistry handlerRegistry(List<Handler<RoutingContext>> handlers) {
+        return new HandlerRegistry(handlers);
+    }
+
+    @Bean
+    public TemplateEngine templateEngine(Vertx vertx) {
+        return HandlebarsTemplateEngine.create(vertx);
+    }
+
+    @Bean
+    public TemplateHandler templateHandler(TemplateEngine templateEngine,
+                                           @Value("${handlers.template.template-dir}") String templateDirectory) {
+        return TemplateHandler.create(templateEngine, templateDirectory, "text/html");
+    }
+
+    @Bean
+    public Router router(Vertx vertx, HandlerRegistry handlerRegistry, TemplateEngine templateEngine) {
+        Router router = Router.router(vertx);
+
+        router.route("/*")
+                .handler(handlerRegistry.get(AddRequestMetricsHandler.class))
+                .handler(
+                        CorsHandler.create(".*.")
+                                .allowCredentials(true)
+                                .allowedHeader("Access-Control-Allow-Method")
+                                .allowedHeader("Access-Control-Allow-Origin")
+                                .allowedHeader("Access-Control-Allow-Credentials")
+                                .allowedHeader("Content-type")
+                                .allowedMethod(HttpMethod.GET)
+                                .allowedMethod(HttpMethod.POST)
+                                .allowedMethod(HttpMethod.OPTIONS));
+
+        router.get("/usync")
+                .handler(handlerRegistry.get(AddDefaultHeadersHandler.class))
+                .handler(handlerRegistry.get(ReadParametersHandler.class))
+                .handler(handlerRegistry.get(ReadSharedIdHandler.class))
+                .handler(handlerRegistry.get(ReadOptOutHandler.class))
+                .handler(handlerRegistry.get(CheckOptOutHandler.class))
+                .handler(handlerRegistry.get(ParseGdprConsentStringHandler.class))
+                .handler(handlerRegistry.get(ReadAuditCookieHandler.class))
+                .handler(handlerRegistry.get(CheckGdprConsentHandler.class))
+                .handler(handlerRegistry.get(CheckCcpaConsentHandler.class))
+                .handler(handlerRegistry.get(CheckVendorHandler.class))
+                .handler(handlerRegistry.get(SetAuditCookieHandler.class))
+                .handler(handlerRegistry.get(SetSharedIdCookieHandler.class))
+                .handler(handlerRegistry.get(RedirectHandler.class))
+                .handler(handlerRegistry.get(RespondDefaultHandler.class));
+
+        router.get("/id")
```

Comment:
Does route `/id` need call `CheckOptOutHandler` ?

Reply:
It doesn't. the id route is exclusively used by the shared id module in prebid to retrieve the value of the cookie and save it to browser storage. The CheckOptOutHandler will short circuit the request to no content if the request is opted out, but module needs to know the value of the cookie at all times, regardless of opt out.

Reply:
@coryhammon1 Thanks for explanation. 

Somehow I mis-marked this comment. 
My concern is not for get route, but for post route: `router.post("/id")` . I marked my concern on wrong place :(

How does post "/id" get called?
Inside `SyncSharedIdHandler`, if new id is: ABC, and existing id is: `cookie.shared-id.opt-out-value=00000000000000000000000000`, then it is possible id: ABC override the opt out id.

If `router.post("/id")` calls `CheckOptOutHandler`, then there is no problem. When it doesn't, we need review if that is correct or not.

Reply:
The requirements for the post id are unique, because it's the interface used by the shared id user module in prebid.js. The timestamp check logic will actually prevent the new id from rewriting the opted out value, but it would be better to have that be explicit. I'll update it now.

Reply:
@coryhammon1 I notice you fix for the issue. I think the idea is correct, just one bug in the fix. The correct code suppose be:
```
        if (newSharedId.equals(sharedIdOptOutValue) || userId.equals(sharedIdOptOutValue)) {
            dataContext.setUserId(sharedIdOptOutValue);
            dataContext.setIsSyncedUserId(true);
            routingContext.next();
            return;
        }
```

Reply:
it is good now.